nix/workbench/manifest.sh

```diff
@@ -115,10 +115,8 @@ manifest_git_checkout_state_desc() {
 }
 
 manifest_cabal_project_dep_pin_hash() {
-    local project_file=$1/cabal.project
-    local dep=$2
-    grep "^[ ]*location: .*/${dep}\$" "${project_file}" -A1 \
-        | tail -n-1 | sed 's/^.* tag: //'
+    # FIXME
+    echo 0123456789ABCDEF0123456789ABCDEF01234567
```

Comment:
Knowing exact gitrevs for key dependencies is useful, since it simplifies discussions around reproduction.

Can we make the `cardanoHaskellPackages` (and I specifically mean the Nix flake) supply the gitrevs for all constituent packages it provides?

Then we could plug in this information here.

Reply:
I maybe missing some context but if you depend on versions, is not even better? The versions are not supposed to be mutable either.

Reply:
I think it would be more appropriate to use the package versions? That should be enough for reproducibility, right? I think @andreabedini had a way to do that with just `ghc-pkg`, so that should be straightforward.

Reply:
In general, this is going to be at git revision granularity -- since we don't always work with released versions of packages.

Think feature branch development, for example.

So, having gitrevs uniformly available would be a consistent approach.

Having versions _in addition_ -- would be nice as well.

Reply:
I tried to update this part but I couldn't really do it without understanding its intended use. I don't think there's a way to push the gitrevs through the index. I suggest we stick to versions. @deepfire, just to make ure I understand, what are gitrevs useful for? looking/finding the right source used? We are thinking of adding html pages for the index, they could include a link to the corresponding source (maybe).

Reply:
> In general, this is going to be at git revision granularity -- since we don't always work with released versions of packages.

right, well, perhaps we can change this script to list both the gitrevs from `source-package-repository` stanzas in `cabal.project` and normal dependencies from the plan. if you add a `source-package-repository` for some in work in progress depedency, they will override the depedency in the repository and show up here.

What do you think?

Reply:
> I don't think there's a way to push the gitrevs through the index.

@angerman, do you think we have no options here?

Reply:
Why do you need the gitrevs for _released_ package versions? Getting them when you're using a `source-repository-package` seems essential indeed, but if you are using a package from the repo, surely the version is sufficient?

Reply:
@michaelpj, again, this is just simpler when you deal with reproductions and general discussion of changes.

When you have the gitrev available, that's a hard source of truth.  Whereas with git tags it's sometimes not exactly a 1-1 relationship with cabal package versions, since it's managed by humans.

Reply:
@andreabedini, so from what @angerman says, the information is available somewhere -- can we make it available in the flake as well?

Reply:
@deepfire its in the toml file in the cardano-Haskell-packages repo.

You need to pin the repo through a nix input anyway, and we could then just read the rev from the toml file in nix from there for the package/version; if needed.

Reply:
> again, this is just simpler when you deal with reproductions and general discussion of changes.

I don't agree with this. Revisions are long, confusing, don't obviously have an ordering. Versions are much easier to deal with 90% of the time. If you're dealing with something unreleased, sure you need revisions, but I still really don't see how a discussion about released versions could be ambiguous when using versions.

Reply:
I never said versions are bad to have!

Only suggesting that losing the gitrevs is unfortunate, because gitrevs are precise and not subject to human errors and other deviations.

Having a cabal version, on the other hand, does not guarantee you a gitrev.

Reply:
@deepfire while it is true that _hackage.haskell.org_ packages do not have revisions, due to the design of hackage allowing arbitrary source management, and a such just release tarball uploads, `cardano-haskell-packages` _does_ have revisions. So at least for `cardano-haskell-packages` you can go from version number to git revision e.g. [measures-0.1.0.0](https://raw.githubusercontent.com/input-output-hk/cardano-haskell-packages/main/_sources/measures/0.1.0.0/meta.toml);

@andreabedini maybe we should ensure that people don't put _tags_ as revisions in the url, as those may change over time (or resolve them at the point of index constructions).

Reply:
@angerman At the moment we are pointing to arbitrary tarballs, when I implement a proprer github backend I can do this.

Reply:
@andreabedini, in the interim it'd be nice to have `cardanoHaskellPackages` provide at least the versions in some kind of a `package-name -> version string` attrset.